oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsSoundCard.java

```diff
@@ -0,0 +1,127 @@
+     * </ul>
+     * @param index
+     *            The row to fetch from
+     * @return The audio card name
+     */
+    private static String getAudioCardCompleteName(int index) {
+        return WmiUtil.getString(AUDIO_CARD_QUERY_RESULT,SoundCardName.MANUFACTURER,index) + " " +
+                WmiUtil.getString(AUDIO_CARD_QUERY_RESULT,SoundCardName.NAME,index);
+    }
+
+    /**
+     * Gets the standard audio card name returned by the Name property of Win32_SoundDevice query.
+     * @param index
+     *          The row to fetch from.
+     * @return  The audio card name.
+     */
+    private static String getAudioCardName(int index) {
+        return WmiUtil.getString(AUDIO_CARD_QUERY_RESULT,SoundCardName.NAME,index);
+    }
+
+    /**
+     * Gets the 'complete' driver name . The 'complete' driver name is infact a concatenated string
+     * of the values returned by the AUDIO_CARD_KERNEL_QUERY query result consisting of :
+     * <ul>
+     *     <li>The DriverName property </li>
+     *     <li>The DriverVersion property</li>
+     *     <li>The DeviceName property</li>
+     *     <li>The DriverProvider property</li>
+     * </ul>
+     * Note: The query of AUDIO_CARD_KERNEL_QUERY is like Get-WmiObject Win32_PnPSignedDriver |
+     * where {$_.devicename -like "*(<b>the name of the card returned by Win32_SoundDevice Name Property</b>)*"}
+     * @param index
+     *        The row to fetch from
+     * @return  The 'complete' name of the driver.
+     */
+    private static String getAudioCardKernelVersion(int index) {
+        String audioCardKernel;
+
+        WbemcliUtil.WmiQuery<SoundCardKernel> cardKernelQuery = new WbemcliUtil.WmiQuery<>("Win32_PnPSignedDriver WHERE DeviceName=\""
+                + AUDIO_CARD_QUERY_RESULT.getValue(SoundCardName.NAME, index)
+                + "\"", SoundCardKernel.class);
+        WbemcliUtil.WmiResult<SoundCardKernel> cardKernelQueryResult = WmiUtil.queryWMI(cardKernelQuery);
+
+        audioCardKernel =
+                WmiUtil.getString(cardKernelQueryResult, SoundCardKernel.DRIVERPROVIDERNAME, index) + " "
+                        + WmiUtil.getString(cardKernelQueryResult, SoundCardKernel.DEVICENAME, index) + " "
+                        + WmiUtil.getString(cardKernelQueryResult, SoundCardKernel.DRIVERNAME, index) + " "
+                        + WmiUtil.getString(cardKernelQueryResult, SoundCardKernel.DRIVERVERSION, index);
+
+        return audioCardKernel;
+    }
+
+    /**
+     * Returns the sound cards..
+     * <br>
+     * NOTE : The reason why the codec name is same as the card name is because windows does not provide
+     * the name of the codec chip but sometimes the name of the card returned is infact the name of the codec
+     * chip also.
+     * Example : Realtek ALC887 HD Audio Device
+     * @return List of sound cards
+     */
+    public static List<WindowsSoundCard> getSoundCards() {
+        List<WindowsSoundCard> cards = new ArrayList<>();
```

Comment:
Suggestion:  populate the map described @ line 94 comment here.   Then query the audio card query and iterate over the results.   You'd pull the version from the map, and calculate the name and complete name from the query result.